analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldVisible.cs

```diff
@@ -46,21 +46,22 @@ public sealed class StaticFieldVisible : SonarDiagnosticAnalyzer
                 c =>
                 {
                     var fieldDeclaration = (FieldDeclarationSyntax)c.Node;
-                    foreach (var field in fieldDeclaration.Declaration.Variables
-                        .Select(variableDeclaratorSyntax => new
-                        {
-                            Syntax = variableDeclaratorSyntax,
-                            Symbol = c.SemanticModel.GetDeclaredSymbol(variableDeclaratorSyntax) as IFieldSymbol
-                        })
-                        .Where(f => FieldIsRelevant(f.Symbol)))
+                    foreach (var diagnostic in GetDiagnostics(fieldDeclaration.Declaration, c.SemanticModel))
                     {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, field.Syntax.Identifier.GetLocation(), field.Syntax.Identifier.ValueText));
+                        c.ReportDiagnosticWhenActive(diagnostic);
                     }
                 },
                 SyntaxKind.FieldDeclaration);
 
+        private static IEnumerable<Diagnostic> GetDiagnostics(VariableDeclarationSyntax declaration, SemanticModel semanticModel) =>
+            declaration.Variables
+                       .Select(declarator => new {Syntax = declarator, Symbol = semanticModel.GetDeclaredSymbol(declarator) as IFieldSymbol})
+                       .Where(f => FieldIsRelevant(f.Symbol))
```

Comment:
The `Symbol` part is used only in the `Where`, so it's causing more syntax and readability trouble than it solves.
Let's move the `GetDeclaredSymbol` part to the `Where` section to simplify the overall processing.
```suggestion
                       .Where(f => FieldIsRelevant(semanticModel.GetDeclaredSymbol(declarator) as IFieldSymbol))
```